### DISCUSSION TITLE
Changing static body to dynamic

### DISCUSSION THREAD

**Discussion by JasperKr** [jrouwe/JoltPhysics#954] [Q&A]
I want to change the motion type of a body that was created as a static object in a non-moving object layer. 

I tried setting the body to dynamic and changing the object layer to a moving one using SetObjectLayer but once i activate the body, it crashes without an error. 

The no error part is probably because I'm using a C binding and it doesn't translate well. However I'm wondering if it's still possible, otherwise I'll probably have to re-create the entire physics body.

**Comment by jrouwe** [comment]
It is certainly possible to switch from static to dynamic, take a look at the example here:

https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/General/ChangeMotionTypeTest.cpp

Pay extra attention to this line:

https://github.com/jrouwe/JoltPhysics/blob/7da4f39fa65c3b592cf3b2496f92417f2d1fe31e/Samples/Tests/General/ChangeMotionTypeTest.cpp#L29

I would also suggest testing with a debug build as there should be asserts that trigger when you're doing something wrong.

**Reply by JasperKr** [reply]
i see. 
I can't edit those values sadly so I'll have to recompile the binding with some extra parameters to the bodyCreationSettingsCreate method.
Thanks for the quick reply!
